ios/ReactNativePageView.m

```diff
@@ -10,412 +9,333 @@
-    pageViewController.delegate = self;
-    pageViewController.dataSource = self;
-    
-    for (UIView *subview in pageViewController.view.subviews) {
-        if([subview isKindOfClass:UIScrollView.class]){
-            ((UIScrollView *)subview).delegate = self;
-            ((UIScrollView *)subview).keyboardDismissMode = _dismissKeyboard;
-            ((UIScrollView *)subview).delaysContentTouches = NO;
+    self.reactPageViewController = [[UIPageViewController alloc] initWithTransitionStyle:self.transitionStyle
+                                                                   navigationOrientation:self.orientation
+                                                                                 options:options];
+    self.reactPageViewController.dataSource = self;
+    self.reactPageViewController.delegate = self;
+    [self addSubview:self.reactPageViewController.view];
+
+    for (UIView *subview in self.reactPageViewController.view.subviews) {
+        if([subview isKindOfClass:UIScrollView.class]) {
             self.scrollView = (UIScrollView *)subview;
+            self.scrollView.delegate = self;
+            self.scrollView.scrollEnabled = self.scrollEnabled;
+            break;
         }
     }
-    
-    self.reactPageViewController = pageViewController;
-    
-    UIPageControl *pageIndicatorView = [self createPageIndicator];
-    
-    pageIndicatorView.numberOfPages = self.reactSubviews.count;
-    pageIndicatorView.currentPage = self.initialPage;
-    pageIndicatorView.hidden = !self.showPageIndicator;
-    
-    self.reactPageIndicatorView = pageIndicatorView;
-    
-    [self reactAddControllerToClosestParent:pageViewController];
-    [pageViewController.view addSubview:pageIndicatorView];
-    [self addSubview:pageViewController.view];
-    
-    pageViewController.view.frame = self.bounds;
-    
-    [self shouldScroll:self.scrollEnabled];
-    
-    if (@available(iOS 9.0, *)) {
-        pageIndicatorView.translatesAutoresizingMaskIntoConstraints = NO;
-        NSLayoutConstraint *bottomConstraint = [pageIndicatorView.bottomAnchor constraintEqualToAnchor: pageViewController.view.bottomAnchor constant:0];
-        NSLayoutConstraint *leadingConstraint = [pageIndicatorView.leadingAnchor constraintEqualToAnchor: pageViewController.view.leadingAnchor constant:0];
-        NSLayoutConstraint *trailingConstraint = [pageIndicatorView.trailingAnchor constraintEqualToAnchor: pageViewController.view.trailingAnchor constant:0];
-        
-        [NSLayoutConstraint activateConstraints:@[bottomConstraint, leadingConstraint, trailingConstraint]];
-    }
-    [pageViewController.view layoutIfNeeded];
-}
 
-- (void)shouldScroll:(BOOL)scrollEnabled {
-    _scrollEnabled = scrollEnabled;
-    if (self.reactPageViewController.view) {
-        self.scrollView.scrollEnabled = scrollEnabled;
-    }
+    [self attachPageIndicator];
 }
 
-- (void)shouldDismissKeyboard:(NSString *)dismissKeyboard {
```

Comment:
why it has been removed ?

Reply:
Seemed unnecessary to me, since js-side already implemented.  I can add it back.

Reply:
I did not notice is it implemented on the js part, hence it is not needed